src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/SnakeCaseUnitTests.cs

```diff
@@ -0,0 +1,51 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public static class SnakeCaseUnitTests
+    {
+        [Fact]
+        public static void ToSnakeCaseTest()
+        {
+            // These test cases were copied from Json.NET.
+            Assert.Equal("url_value", ConvertToSnakeCase("URLValue"));
+            Assert.Equal("url", ConvertToSnakeCase("URL"));
+            Assert.Equal("id", ConvertToSnakeCase("ID"));
+            Assert.Equal("i", ConvertToSnakeCase("I"));
+            Assert.Equal("", ConvertToSnakeCase(""));
+            Assert.Null(ConvertToSnakeCase(null));
+            Assert.Equal("person", ConvertToSnakeCase("Person"));
+            Assert.Equal("i_phone", ConvertToSnakeCase("iPhone"));
+            Assert.Equal("i_phone", ConvertToSnakeCase("IPhone"));
+            Assert.Equal("i_phone", ConvertToSnakeCase("I Phone"));
+            Assert.Equal("i__phone", ConvertToSnakeCase("I  Phone"));
```

Comment:
I don't think this behavior is correct. IMO all symbols should be treated as whitespaces, and then multiple consecutive whitespaces should be merged into one whitespace, and last remove the heading and tailing whitespaces. 

eg.
```
I  Phone -> i_phone
SnA__ kEcAsE -> sn_a_k_ec_as_e
Hi!! This is text. Time to test. -> hi_this_is_text_time_to_test
__he     llo__u^-^ -> he_llo_u
```

For testing: https://textedit.tools/snakecase

Reply:
That seems reasonable - I *was* considering collapsing underscores, but I was torn whether the user would expect `i__phone` or `i_phone` when there was explicitly multiple spaces.

Reply:
I personally would expect `i_phone` in this case.

Reply:
I've changed this case locally, but I'm waiting for a response on the number issue before I commit the changes, since I'm already tweaking those cases.